internal/convert.go

```diff
@@ -37,11 +38,12 @@ type Conv struct {
 	Location                 *time.Location // Timezone (for timestamp conversion).
 	sampleBadRows            rowSamples     // Rows that generated errors during conversion.
 	Stats                    stats
-	TimezoneOffset           string              // Timezone offset for timestamp conversion.
-	TargetDb                 string              // The target database to which HarbourBridge is writing.
-	UniquePKey               map[string][]string // Maps Spanner table name to unique column name being used as primary key (if needed).
-	SchemaConversionDuration time.Duration       `json:"-"` // Duration of schema conversion.
-	DataConversionDuration   time.Duration       `json:"-"` // Duration of data conversion.
+	TimezoneOffset           string                  // Timezone offset for timestamp conversion.
+	TargetDb                 string                  // The target database to which HarbourBridge is writing.
+	UniquePKey               map[string][]string     // Maps Spanner table name to unique column name being used as primary key (if needed).
+	SchemaConversionDuration time.Duration           `json:"-"` // Duration of schema conversion.
+	DataConversionDuration   time.Duration           `json:"-"` // Duration of data conversion.
+	MigrationData            migration.MigrationData `json:"-"` // Migration related metadata to be sent to spanner.
```

Comment:
Using shared object like conv all over the codebase is an anti pattern and is a huge tech debt in harbourbridge. Lets not add any more fields to it. Here's how we should do this:
1. Create a new file usage.go or metrics.go and have a GetMigrationMetadata method (just need to move PopulateMigrationMetadata from utils.go) which should use other files' methods to create migrationmetadata. Avoid using conv members directly in this method as well as much as possible i.e. by using getters
2. Call this method wherever we need to add this to the context.
Once we make these changes, the changes in this PR would mostly limit to conversion.go and usage/metrics.go.

Reply:
Your point makes sense but just a small question, we will need to pass same request id for all data and schema writes for a single migration so if we can the method to get migration data in conversion.go while sending it with metadata we may end up with different request ids unless we store that in conv object.